### PR TITLE
[13][IMP] Backport job.related.mixin refactoring

### PR DESCRIPTION
**Pull request by hailangvn**
This is to backport refactoring from https://github.com/OCA/connector-interfaces/pull/78/commits/6c25ab2d6c961baf7484ccf87e52522b3655b5ef

### CODE REVIEW

File: connector_importer/models/job_mixin.py [hunk 1]
```diff
@@ -2,17 +2,16 @@
...
 
-class JobRelatedMixin(object):
-    """Mixin klass for queue.job relationship.
+class JobRelatedMixin(models.AbstractModel):
+    """ Mixin klass for queue.job relationship. """
 
-    We do not use an abstract model to be able to not re-define
-    the relation on each inheriting model.
-    """
+    _name = "job.related.mixin"
```
Comment (simahawk) [inline connector_importer/models/job_mixin.py:13]: not used anywhere apparently :thinking:

**Review by simahawk — CHANGES_REQUESTED** [review]

File: connector_importer/models/job_mixin.py (hunk 1 above)
Comment (hailangvn) [reply connector_importer/models/job_mixin.py:13]: Good day @simahawk, `ImportRecord` and `ImportRecordSet` are using it. Please advise if anything need to be changed.
https://github.com/OCA/connector-interfaces/blob/3c9d462a3ef024f7415206d53adfb0c7a7d35d27/connector_importer/models/record.py#L17
https://github.com/OCA/connector-interfaces/blob/3c9d462a3ef024f7415206d53adfb0c7a7d35d27/connector_importer/models/recordset.py#L18

Comment (simahawk) [reply connector_importer/models/job_mixin.py:13]: yes but that's a normal python inheritance which does not need the mixin to inherit from the abstract model.
If you define it as an abstract model I would expect to be used for inheritance in the odoo way.
When I worked on this back in the days there was something that was not working properly.
I'll check.

**Comment by simahawk** [comment]
please check what I've done here https://github.com/OCA/connector-interfaces/pull/82/commits/85dcd1adce32993bdd8418ef29b23fecc90e78f2 ;)

**Pull request closed without merging**